### ISSUE TITLE
Version 2.1, conflict between mount IP and taking pictures

### ISSUE THREAD

**Issue by farn32** [mworion/MountWizzard4#200]
As soon as I enter an IP Adress for the mount in the menu settings / mount, there is no way for taking pictures anymore. When I clear the field with the IP Adress I can get pictures again.

As you can see in the pictures when the mount is connected there aren't any mount parameters in mountwizzard (RA, DEC, etc). I don't now why is this?
Of course as you can see I use an old mount driver (2.15.1). I use this one because it works very well guided or unguided and I red alot of ugly things about the new drivers. Before version 2.1 I didn't have any problems.
Do I have to change the driver?

![image](https://user-images.githubusercontent.com/94228091/141646517-70e22188-ccb8-4fd8-88ad-33348d2b4239.png)

![image](https://user-images.githubusercontent.com/94228091/141646528-c45b9c67-80c5-4542-84cb-0e3dea4de320.png)

**Comment by mworion** [comment]
Hi Fritz,
could you add the log file as well (ideally logging set to debug under settings / misc)
Michel

**Comment by mworion** [comment]
Add about other remark:
You're not talking of a driver, it is the overall firmware of the mount itself. I also heard about some issues in later firmwares, but I never experienced one myself. But It's a decision on your own if you would like to change. 
As Info: as long as you stay on 2.x you should be able to go back. Once you upgraded to 3.x, your will not get back to 2.x versions.
For MW4 I recommend a 2.16.x as there are many functions not available in earlier versions. But there is not need to get to the latest one (3.x).
Hop this helps.

**Comment by farn32** [comment]
Hi Michel 

Sorry, of course it's the firmware of the mount not a driver. 
I don't know about the firmware change maybe the 3.0.8 would be possible, because the guiding changements and the balance issues should be restored to the 2.x versions.

I did the following:

run mountwizzard
connecting the camera and the filterwheel
taking two pictures (works well)
enter the mount IP
taking two pictures (have to abort it, because they cannot be saved)
delete the mount ip adress
taking two pictures (works well)

[mw4-2021-11-13.log](https://github.com/mworion/MountWizzard4/files/7532867/mw4-2021-11-13.log)

Thanks alot for you time

Fritz

**Comment by mworion** [comment]
Hi Fritz,
please update to 2.1.2, which should fix the issue. Regarding the mount connection (which was to problem): I see a lot of connection issues. So something must be wrong (wireless connection, stability?).
Regarding the firmwares: please hav a look to the 2.16.13 (latest 2.x) which could be OK for you as well. But please check your preferences.
Michel

**Comment by farn32** [comment]
Hi Michel

Thanks alot for your help. I think, that I will change the firmware to 2.16.13 as you mentioned.  For the update to mountwizzard I tried with the update script, but 2.1.2 is not available. Do I have to update without the script.

Fritz

**Comment by mworion** [comment]
2.1.2 will need about 30 min as build takes some time

**Comment by farn32** [comment]
Thanks alot Michel I installed the mount firmware 2.16.13 and the new version of mountwizzard. Everything seems to work again.
Thanks alot for your help and the great work.
By the way I think, that I installed the newest version but it says 2.1.0.

![image](https://user-images.githubusercontent.com/94228091/141694488-41b327ef-28e8-4251-b3ac-93bc1c04b418.png)

![image](https://user-images.githubusercontent.com/94228091/141694505-c225e914-5e82-4b93-9d8e-5f8cb8f4f810.png)


All the best

Fritz

**Comment by mworion** [comment]
Hi Fritz,
this is strange. Could you post a log from an update run ?
Michel

**Comment by farn32** [comment]
Hi Michel

Here's the log.

[update.log](https://github.com/mworion/MountWizzard4/files/7540417/update.log)

Fritz

**Comment by alexxw24** [comment]
Hi Michel,

great piece of software. It worked all fine but lately upraded my mount to 3.0.8 and since then taking picture with DSLR isnt possible anymore. Strange enough with my ZWO ASI 120MC guiding cam it still does.

Have tested it on an Intel NUC running ubuntu 20.04 as well as on an Raspi 4 both not working. On a Win10 notebook/laptop with the ASCOM.DSLR driver it works.

Taking out the mounts ip didnt made a change on the linux systems neither.

Please find enclosed latest log. As I understood downgrading from 3.x mount firmware back to 2.x isnt possible so would be great if there is some fix.
  
Thanks for looking at it.

Alex

**Comment by alexxw24** [comment]
[link to latest log file](https://hidrive.ionos.com/lnk/EDuRDjkt)

**Comment by mworion** [comment]
Hi Alex,
I would sort your issue out with a separate issue number. Could you open one (or we use the one you opened recently). It's not the same than from Fritz. 

Basically the firmware version does not have anything related to imaging (beside having a mount connection). 
Michel

**Comment by mworion** [comment]
Fritz,
I checke the update log. For some reason you are not working with MW4 in a virtual environment. This is not OK and something went wrong. This also explains, why you upgrade, but still do not see the actual version number.
Could you explain, how you installed MW4 first and how you did the upgrade ? Did you use the scripts? Which version ? If you still have the install log, please post it as well. For further tests, please open a command window and change directory to you working directory. Then please try the next commands and post the output:

dir (just to get the contents)
python3 --version
python3 -m pip list

if the is a venv directory in your work folder present:

./venv/Scripts/activate

-> then there should be a (venv) prefix in command window

python3 --version
python3 -m pip list

Thanks,
Michel

**Comment by farn32** [comment]
Hi Michel

I did a new install, first python 3.9.8 and then I used the scripts I downloaded from the link:
**MountWizzard4/support/Windows_Scripts.zip**

**Install Log:**
[install.log](https://github.com/mworion/MountWizzard4/files/7574709/install.log)

**output from the commands:**

_dir_

 Verzeichnis von C:\Users\fritz.arn\Desktop\MWWork

19.11.2021  07:05    <DIR>          .
19.11.2021  07:05    <DIR>          ..
09.11.2021  23:19            27’648 CFF132_G316200_MW4.cfg
09.11.2021  23:09            27’648 CFF132_G316200_MW4_.cfg
10.11.2021  18:23    <DIR>          config
14.11.2021  13:02    <DIR>          data
14.11.2021  18:05    <DIR>          image
09.11.2021  23:03             9’385 install.log
09.11.2021  23:04    <DIR>          measure
09.11.2021  23:04    <DIR>          model
13.11.2021  23:27            51’613 mw4-2021-11-13.log
14.11.2021  19:55           241’034 mw4-2021-11-14.log
15.11.2021  18:11             6’568 mw4-2021-11-15.log
19.11.2021  07:06            20’160 mw4-2021-11-19.log
09.11.2021  22:39             3’553 MW4_Install.bat
09.11.2021  22:39             2’077 MW4_Run.bat
09.11.2021  22:39             2’352 MW4_Update.bat
19.11.2021  07:05                 0 run.log
09.11.2021  23:04    <DIR>          temp
19.11.2021  07:05             7’167 update.log
09.11.2021  23:00    <DIR>          venv
09.11.2021  22:40    <DIR>          __MACOSX
              12 Datei(en),        399’205 Bytes
              10 Verzeichnis(se), 381’045’686’272 Bytes frei


_python --version_

Python 3.9.8


_python -m pip list_

C:\Users\fritz.arn\Desktop\MWWork>python -m pip list
Package            Version
------------------ ---------
astropy            4.3.1
certifi            2021.10.8
charset-normalizer 2.0.7
colour-demosaicing 0.1.6
colour-science     0.3.16
comtypes           1.1.10
cycler             0.11.0
deepdiff           5.5.0
idna               3.3
imageio            2.9.0
importlib-metadata 4.6.4
jplephem           2.16
kiwisolver         1.3.2
matplotlib         3.4.2
mountwizzard4      2.1.2
numpy              1.21.2
ordered-set        4.0.2
Pillow             8.3.1
pip                21.2.4
pybase64           1.2.0
pyerfa             2.0.0
pyparsing          3.0.5
PyQt3D             5.15.4
PyQt3D-Qt5         5.15.2
PyQt5              5.15.4
PyQt5-Qt5          5.15.2
PyQt5-sip          12.9.0
python-dateutil    2.8.2
pywin32            301
pywinauto          0.6.8
qimage2ndarray     1.8.3
requests           2.26.0
requests-toolbelt  0.9.1
scipy              1.7.1
sep                1.2.0
setuptools         58.1.0
sgp4               2.20
six                1.16.0
skyfield           1.39
urllib3            1.26.7
wakeonlan          2.0.1
websocket-client   1.2.1
wheel              0.37.0
zipp               3.6.0
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the 'C:\Users\fritz.arn\AppData\Local\Programs\Python\Python39\python.exe -m pip install --upgrade pip' command.



C:\Users\fritz.arn\Desktop\MWWork\venv\Scripts>_activate_

(venv) C:\Users\fritz.arn\Desktop\MWWork\venv\Scripts>_python --version_
Python 3.9.8

(venv) C:\Users\fritz.arn\Desktop\MWWork\venv\Scripts>



(venv) C:\Users\fritz.arn\Desktop\MWWork\venv\Scripts>_python -m pip list_
Package            Version
------------------ ---------
astropy            4.3.1
certifi            2021.10.8
charset-normalizer 2.0.7
colour-demosaicing 0.1.6
colour-science     0.3.16
comtypes           1.1.10
cycler             0.11.0
deepdiff           5.5.0
idna               3.3
imageio            2.9.0
importlib-metadata 4.6.4
jplephem           2.16
kiwisolver         1.3.2
matplotlib         3.4.2
mountwizzard4      2.1.0
numpy              1.21.2
ordered-set        4.0.2
Pillow             8.3.1
pip                21.3.1
pybase64           1.2.0
pyerfa             2.0.0
pyparsing          3.0.5
PyQt3D             5.15.4
PyQt3D-Qt5         5.15.2
PyQt5              5.15.4
PyQt5-Qt5          5.15.2
PyQt5-sip          12.9.0
python-dateutil    2.8.2
pywin32            301
pywinauto          0.6.8
qimage2ndarray     1.8.3
requests           2.26.0
requests-toolbelt  0.9.1
scipy              1.7.1
sep                1.2.0
setuptools         58.5.3
sgp4               2.20
six                1.16.0
skyfield           1.39
urllib3            1.26.7
wakeonlan          2.0.1
websocket-client   1.2.1
wheel              0.37.0
zipp               3.6.0

**Comment by mworion** [comment]
Hi Fritz,
as you could see, you have two different MW4 versions installed and we have still the MW4 install in the system itself.
Normally if you deinstall python and python launcher and reinstall python (not just overinstall), the system packages should be removed and you will have a clean install. If you once made a MW4 install manually with 

pip install mountwizzard4

without having the virtual environment enabled (see prefix), you have the install as systemwide done - not good.
Please remove python and python launcher and reinstall python. The command python -m pip list should show not more than 3-5 entries, never more.

If you test different installations (you could have more than one), please use a separate workfolder for each install. Please do not do the new version inside an actual worker. You might mix up the script locations and the install they reference. So please a clean set: worker with scripts and all what will be installed and generated during runtime. If you would like to share configurations, just copy them from subdir to subdir (please do not  copy the venv folder)

After the install script, python -m pip list still will show these 3-5 items max. So in set system there will be no install at all.
If you activate the venv with the command activate (prefix shown), the same command python -m pip list will show you the full install.

Michel

**Comment by farn32** [comment]
Hi Michel thans alot for your help.
This is a bit too much to unserstand because of this python thing. OK in short I deinstall all the python stuff and then I'll do a fresh install according your installation manual:

python and mountwizzard, with the scripts, is that right?

Your manual says the following:

Warning
Windows makes a hard split between 32bit and 64bit versions. If your drivers and setup uses 64bit solutions, please install 64bit python!

Depending on your Windows version please download or directly run the web installer from:

https://www.python.org/downloads/windows/

and follow the installation procedure.

Warning
Please take care that during the installation the checkbox “Add Python Path” is selected and to install for a single user if you want to use the scripts.

Sorry for that.

Fritz

**Comment by mworion** [comment]
Fritz,

If you want, we could do a teamviewer session ?
Michel

Von meinem iPad gesendet

**Comment by farn32** [comment]
Hi Michel,

Thanks alot for the offer, I'll try it and when it won't work I'll contact you again.

I really want to steal your time :)

I report back...

Thanks alot for the great support.

Fritz

**Comment by farn32** [comment]
Hi Michel

I did the reinstall and I think, that everything looks good. The command python -m pip gives more than 3 to 5 entries, about the same as before.
At the moment the newest version 2.1.2 is installed.

I tried with the moravian camera and the connected mount and it works with taking pictures.

[install.log](https://github.com/mworion/MountWizzard4/files/7575062/install.log)

[update.log](https://github.com/mworion/MountWizzard4/files/7575063/update.log)

Thanks alot for your help.

Fritz
